google/cloud/aiplatform/_featurestores/featurestore.py

```diff
@@ -0,0 +1,457 @@
+                   ``foobarfeature``).
+                -  ``feature_id: foo AND description: bar`` --> Matches
+                   a Feature with ID containing the substring ``foo``
+                   and description containing the substring ``bar``.
+
+                Besides field queries, the following exact-match filters
+                are supported. The exact-match filters do not support
+                wildcards. Unlike field-restricted queries, exact-match
+                filters are case-sensitive.
+
+                -  ``feature_id``: Supports = comparisons.
+                -  ``description``: Supports = comparisons. Multi-token
+                   filters should be enclosed in quotes.
+                -  ``entity_type_id``: Supports = comparisons.
+                -  ``value_type``: Supports = and != comparisons.
+                -  ``labels``: Supports key-value equality as well as
+                   key presence.
+                -  ``featurestore_id``: Supports = comparisons.
+
+                Examples:
+
+                -  ``description = "foo bar"`` --> Any Feature with
+                   description exactly equal to ``foo bar``
+                -  ``value_type = DOUBLE`` --> Features whose type is
+                   DOUBLE.
+                -  ``labels.active = yes AND labels.env = prod`` -->
+                   Features having both (active: yes) and (env: prod)
+                   labels.
+                -  ``labels.env: *`` --> Any Feature which has a label
+                   with ``env`` as the key.
+
+                This corresponds to the ``query`` field
+                on the ``request`` instance; if ``request`` is provided, this
+                should not be set.
+            page_size (int):
+                Optional. The maximum number of Features to return. The
+                service may return fewer than this value. If
+                unspecified, at most 100 Features will be
+                returned. The maximum value is 100; any value
+                greater than 100 will be coerced to 100.
+            project (str):
+                Optional. Project to list features in. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to list features in. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to list features. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            List[Features] - A list of managed feature resource objects
+        """
+        resource = cls._empty_constructor(
+            project=project, location=location, credentials=credentials
+        )
+
+        # Fetch credentials once and re-use for all `_empty_constructor()` calls
+        creds = initializer.global_config.credentials
```

Comment:
This should default to the credentials passed in then the `global_config` credentials. This is handled by the construction of the resource.

```suggestion
        creds = resource.credentials
```